packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -755,21 +755,15 @@ class ServiceExtensionManager extends Disposer {
           // Isolate has changed again.
           return;
         }
-        for (String extension in isolate.extensionRPCs) {
-          await _maybeAddServiceExtension(extension);
-          if (isolateRef != _mainIsolate.value) {
-            // Isolate has changed again.
-            return;
-          }
-        }
+        await Future.wait([
```

Comment:
Without this we had super confusing bugs as our app state would reach an async gap with only some of the service extensions added. Now they are all added at the same block.